### ISSUE TITLE
Contact Form Submission - Suggestion (Silver - More on Prefix Sums)

### ISSUE THREAD

**Issue by maggieliu05** [cpinitiative/usaco-guide#4916]
Someone submitted the contact form!

**URL**: https://usaco.guide/silver/more-prefix-sums
**Module**: Silver - More on Prefix Sums
**Topic**: Suggestion
**Message**: 
I don't believe "Nusret Gokce" problem belongs in the exercices section because it doesn't use prefixsum

**Comment by bqi343** [comment]
agreed. though an argument can be made that the solution is similar in structure to the maximum subarray sum solution

**Comment by QuicksVolex1** [comment]
As a beginner, i would really appreciate if you can explain how that argument can be made. Since i'm struggling to even convince myself the solution works i think that would be really beneficial.

**Comment by bqi343** [comment]
it's as the editorial describes. you iterate over the array in one direction keeping a running max, then the other.

```py
N, M = map(int, input().split())

S = list(map(int, input().split()))

running_max = 0
for i in range(N):
	running_max = max(running_max - M, S[i])
	S[i] = running_max

running_max = 0
for i in reversed(range(N)):
	running_max = max(running_max - M, S[i])
	S[i] = running_max

print(" ".join(map(str, S)))
```

**Comment by bqi343** [comment]
actually, I'll keep this open because this problem needs an internal solution
